### PR TITLE
Add missing doc to the Metrics APIs and TimeOnly type.

### PR DESCRIPTION
**Pull request by tarekgh**
https://github.com/dotnet/runtime/issues/72409
https://github.com/dotnet/runtime/issues/72404

### CODE REVIEW

**Comment by tarekgh** [comment]
@carlossanlop could you please have a look?

File: xml/System/TimeOnly.xml [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ If `start` and `end` are equal, this method returns `false`, meaning there is ze
         <Parameter Name="provider" Type="System.IFormatProvider" Index="1" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
+        <param name="s">A span of characters to parse.</param>
         <param name="provider">To be added.</param>
```
Comment (carlossanlop) [inline xml/System/TimeOnly.xml:1005]: This param is also missing documentation.

**Review by carlossanlop — COMMENTED** [review]
Thanks, @tarekgh . I left some suggestions.

File: xml/System/TimeOnly.xml [hunk 2]
```diff
@@ -1773,9 +1773,9 @@ The accepted standard formats are 'r', 'R', 'o', 'O', 't', and 'T'.
         <Parameter Name="result" Type="System.TimeOnly" RefType="out" Index="2" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <param name="provider">To be added.</param>
-        <param name="result">To be added.</param>
+        <param name="s">A span of characters to parse.</param>
+        <param name="provider">An object that provides culture-specific formatting information about <paramref name="s" />.</param>
+        <param name="result">When this method returns, contains the TimeOnly value equivalent to the time contained in <paramref name="s" />, if the conversion succeeded, or MinValue if the conversion failed. The conversion fails if <paramref name="s" /> is the empty string or does not contain a valid string representation of a time. This parameter is passed uninitialized.</param>
```
Comment (carlossanlop) [inline xml/System/TimeOnly.xml:1778]:
```suggestion
        <param name="result">When this method returns, contains the <see cref="T:System.TimeOnly" /> value equivalent to the time contained in <paramref name="s" />, if the conversion succeeded, or <see cref="P:System.TimeOnly.MinValue" /> if the conversion failed. The conversion fails if <paramref name="s" /> is the empty string or does not contain a valid string representation of a time. This parameter is passed uninitialized.</param>
```

File: xml/System/TimeOnly.xml [hunk 3]
```diff
@@ -1810,11 +1810,12 @@ The accepted standard formats are 'r', 'R', 'o', 'O', 't', and 'T'.
...
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <param name="provider">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="s">A string to parse.</param>
+        <param name="provider">An object that provides culture-specific formatting information about <paramref name="s" />.</param>
+        <param name="result">When this method returns, contains the TimeOnly value equivalent to the time contained in <paramref name="s" />, if the conversion succeeded, or MinValue if the conversion failed. The conversion fails if <paramref name="s" /> is the empty string or does not contain a valid string representation of a time. This parameter is passed uninitialized.</param>
```
Comment (carlossanlop) [inline xml/System/TimeOnly.xml:1815]:
```suggestion
        <param name="result">When this method returns, contains the <see cref="T:System.TimeOnly" /> value equivalent to the time contained in <paramref name="s" />, if the conversion succeeded, or <see cref="P:System.TimeOnly.MinValue" /> if the conversion failed. The conversion fails if <paramref name="s" /> is the empty string or does not contain a valid string representation of a time. This parameter is passed uninitialized.</param>
```

**Review by carlossanlop — APPROVED** [review]
Two more additional suggestions to use cref. Aside from that, LGTM.

**Pull request merged**